### ISSUE TITLE
excited to get this up and running but appear to need more info

### ISSUE THREAD

**Issue by greggdourgarian** [runtimeverification/iele-semantics#31]
looks like this requires ghc 8.2.2...which is only avl on debian?   would like an eli5 on how to install, tx

**Comment by dwightguth** [comment]
Working on it right now... hope to have something by the end of the day.

**Comment by bmmoore** [comment]
For building the assembler, the stack tool should manage ghc versions for you, at least on any relatively recent Linux distribution.
@dwightguth is working out instructions for installing from scratch on Ubuntu. (installing ocaml dependencies for compiling the K definitions also needs instructions).

**Comment by greggdourgarian** [comment]
tx!

**Comment by dwightguth** [comment]
If you still need more help, I suggest looking at the installation instructions in the install branch that I just pushed. It should be merged sometime on Monday but you can get a head start trying it out and testing it if you want. I'm pretty sure it's ready and tested but if you find issues just let us know.

**Comment by greggdourgarian** [comment]
Below is the output from running the install instructions on a virgin ubuntu14.04.5 x32.  I tried to get by the various path errors and eventually had "make: *** No rule to make target 'deps'.  Stop." that i couldn't get by.

Setting up libffi-dev:i386 (3.1~rc1+r3.0.13-12ubuntu0.2) ...
Processing triggers for libc-bin (2.19-0ubuntu6.13) ...

Using generic bindist...

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0   619    0     0   2719      0 --:--:-- --:--:-- --:--:--  2719
100 11.0M  100 11.0M    0     0  17.8M      0 --:--:-- --:--:-- --:--:-- 56.9M
Installing Stack to: /usr/local/bin/stack...

-------------------------------------------------------------------------------

Stack has been installed to: /usr/local/bin/stack

WARNING: '/root/.local/bin' is not on your PATH.
    For best results, please add it to the beginning of PATH in your profile.

root@ubuntu-512mb-nyc3-01:~# cd .build/secp256k1 && ./autogen.sh && ./configure --enable-module-recovery && make && sudo make install # install secp256k1 from bitcoin-core
-bash: cd: .build/secp256k1: No such file or directory
root@ubuntu-512mb-nyc3-01:~# make deps # Build dependencies not installed by package manager
make: *** No rule to make target `deps'.  Stop.
root@ubuntu-512mb-nyc3-01:~# eval `opam config env` # add OCAML installation to path
The program 'opam' is currently not installed. You can install it by typing:
apt-get install opam

**Comment by dwightguth** [comment]
I have several questions:

1. What branch are you trying to build? You should be trying to build the install branch for the time being until it gets merged.
2. Did you run `git submodule update --init`?
3. Are you sure you installed the correct list of dependencies? It seems that opam is not installed.

**Comment by greggdourgarian** [comment]
hi...thanks for your help

i have retried it with the install branch and got much further...however towards the end i got the following error 

[ERROR] curl: code 502 while downloading http://www-verimag.imag.fr/~monniaux/download/mlgmp_20120224.tar.gz
[ERROR] The sources of the following couldn't be obtained, aborting:
          - mlgmp.20150824
        (This may be fixed by running 'opam update')
Makefile:99: recipe for target 'deps' failed

And then when running the Make, i got: 

ocamlfind opt -c .build/rvk/ethereum-kompiled/constants.ml -package gmp -package zarith
ocamlfind: Package `gmp' not found
Makefile:108: recipe for target '.build/rvk/ethereum-kompiled/interpreter' failed
make: *** [.build/rvk/ethereum-kompiled/interpreter] Error 2

**Comment by dwightguth** [comment]
This appears to be an upstream error from us. The owner of that package appears to be having server issues. I should have a copy of the tarball locally on my workstation, so when I get in to work tomorrow I will see about getting you an updated version of the package that no longer tries to download it from that server.

**Comment by dwightguth** [comment]
@greggdourgarian I am no longer getting the server error when I try to download this file. Can you try again?

**Comment by greggdourgarian** [comment]
hi Dwight...looks like it built fine...tx for sticking w me

**Comment by dwightguth** [comment]
I'm closing this. Feel free to reopen if you have more trouble.
